src/syntax_mapping.rs

```diff
@@ -24,6 +24,12 @@ impl<'a> SyntaxMapping<'a> {
         let mut mapping = Self::empty();
         mapping.insert("*.h", MappingTarget::MapTo("C++")).unwrap();
         mapping.insert("*.fs", MappingTarget::MapTo("F#")).unwrap();
+        mapping
+            .insert("*.js", MappingTarget::MapTo("JavaScript (Babel)"))
```

Comment:
Ideally, we should be able to implement this feature without adding theses new syntax mappings. Instead of just querying the syntax_mapping, we would need to run the full syntax detection on the `test.xx` filename.

Reply:
Ops, I was just following the similar commits before and thought it would do 😅, should I revert it and change the implementation?

Reply:
Yes, please revert this.

By "full syntax detection", I mean: determine the syntax in the same way that `bat` usually does this for a normal file. This involves not just checking the syntax mapping, but also the full database of syntaxes. See `src/assets.rs` => `HighlightingAssets::get_syntax`.